chain-signatures/node/src/rpc.rs

```diff
@@ -645,3 +691,76 @@ async fn try_publish_eth(
     }
     // TODO: add metrics for Ethereum
 }
+
+use chain_signatures_project::instruction::Respond as SolanaRespond;
+use chain_signatures_project::Signature as SolanaContractSignature;
+use chain_signatures_project::AffinePoint as SolanaContractAffinePoint;
+use chain_signatures_project::accounts::Respond as SolanaRespondAccount;
+use solana_sdk::signature::Signer as SolanaSigner;
+async fn try_publish_sol(
+    sol: &SolanaClient,
+    action: &PublishAction,
+    timestamp: &Instant,
+    signature: &Signature,
+) -> Result<(), ()> {
+    let program = sol.client.program(sol.program_id).map_err(|_| ())?;
+    
+    let request_ids = vec![action.request.indexed.id.request_id];
+    let signature = SolanaContractSignature {
+        big_r: SolanaContractAffinePoint {
+            x: signature.big_r.to_encoded_point(false).as_bytes()[1..33].try_into().unwrap(),
+            y: signature.big_r.to_encoded_point(false).as_bytes()[33..65].try_into().unwrap(),
+        },
+        s: signature.s.to_bytes().into(),
+        recovery_id: signature.recovery_id,
+    };
+    
+    let tx = program
+        .request()
+        .signer(sol.payer.clone())
+        .accounts(SolanaRespondAccount {
+            responder: sol.payer.clone().try_pubkey().unwrap(),
```

Comment:
Should we return an error rather than unwrap?

Reply:
It is a configuration error (when sol key, that used by node to respond is not provided or incorrect at startup), so I chose to end the node.

Reply:
we should probably make the types more concrete then in the future, such that this unwrap() isn't a code smell later or just in case a node messes up their setup and no SOL related tx makes it in for a while